gateway/channel_users.go

```diff
@@ -0,0 +1,126 @@
+package gateway
+
+import "github.com/nlopes/slack"
+
+// getChannelUsersFromAPI queries the Slack API for a list of users in the given channel, returning
+// SlackUser objects for each one
+func (sc *SlackClient) getChannelUsersFromAPI(channelID string) (users []*SlackUser, err error) {
+	hasMore := true
+	guicp := &slack.GetUsersInConversationParameters{
+		ChannelID: channelID,
+		Limit:     1000,
+	}
+	for hasMore {
+		var userIDs []string
+		userIDs, guicp.Cursor, err = sc.client.GetUsersInConversation(guicp)
+		if err != nil {
+			return
+		}
+
+		for _, userID := range userIDs {
+			var user *SlackUser
+			user, err = sc.ResolveUser(userID)
+			if err != nil {
+				return
```

Comment:
Likewise here, though I think it matters less here (if we can't resolve a user, then something kinda bad happened)